File: TaskMasterAPI/src/main/java/com/devlab/taskmaster/dto/user/UserTaskDTO.java
package com.devlab.taskmaster.dto.user;

import com.devlab.taskmaster.enumeration.TaskPriority;
import com.devlab.taskmaster.enumeration.TaskStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserTaskDTO {

    private Integer id;

    private String name;

    private TaskStatus status;

    private TaskPriority priority;
}

File: TaskMasterWeb/TaskMasterWeb.Client/Dtos/User/UserTaskDto.cs
namespace TaskMasterWeb.Client.Dtos.User;

public record UserTaskDto
{
    public int Id { get; set; }

    public string Name { get; set; } = default!;

    public string Status { get; set; } = default!;

    public string Priority { get; set; } = default!;
}